osquery/sql/virtual_sqlite_table.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+
+#include "osquery/sql/sqlite_util.h"
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+
+Status genSqliteQueryRow(sqlite3_stmt* stmt,
+                         QueryData& qd,
+                         const fs::path& sqlite_db) {
+  Row r;
+  for (auto i{0}; i < sqlite3_column_count(stmt); ++i) {
```

Comment:
nit: my own personal preference here, but I feel like the braced initialization is a bit excessive. Why not leave the typing and equal here as a normal `for` loop? Am I coming off as too old and fighting the future? :)

Reply:
I mean technically I think this should be: `decltype(sqlite3_column_count(stmt)) i{0};`